### PR TITLE
Upgrade sqlite amalgamation to latest 3.12.0

### PR DESCRIPTION
**Pull request by tomkwok**

### CODE REVIEW

File: code/sqlite3ext.h [hunk 1]
```diff
@@ -17,7 +17,7 @@
 */
 #ifndef _SQLITE3EXT_H_
 #define _SQLITE3EXT_H_
-#include "sqlite3-binding.h"
+#include "sqlite3.h"
```
Comment (mattn) [inline code/sqlite3ext.h]: should be fixed.

**Comment by tomkwok** [comment]
Done.

**Comment by evanj** [comment]
I was looking into this due to the better defaults in 3.12 (bigger page size), as well as the usual bug fixes. At this point: 3.12.2 has been released, so it would make sense to update this branch to that before merging probably?

**Comment by mattn** [comment]
Sorry about I can't take a time to looking this. It seems newer exists.

https://github.com/mattn/go-sqlite3/pull/304

I'll merge #304 . Sorry.

**Pull request closed without merging**